coverage/PENDING_SUBMIT/amdvlk/0545-COVERAGE-instructions-h-2677/cov-do-while-negative-iterator-nested-loops-increment-array-element.amber

```diff
@@ -0,0 +1,470 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific LLVM code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# #define _int_5 _GLF_uniform_int_values[0]
+# #define _int_0 _GLF_uniform_int_values[1]
+# #define _int_20 _GLF_uniform_int_values[2]
+# #define _int_1 _GLF_uniform_int_values[3]
+# #define _int_2 _GLF_uniform_int_values[4]
+# #define _int_3 _GLF_uniform_int_values[5]
+# #define _int_4 _GLF_uniform_int_values[6]
+# #define _int_19 _GLF_uniform_int_values[7]
+# #define _int_16 _GLF_uniform_int_values[8]
+# #define _int_17 _GLF_uniform_int_values[9]
+# #define _int_18 _GLF_uniform_int_values[10]
+# #define _float_1_0 _GLF_uniform_float_values[0]
+#
+# precision highp int;
+# precision highp float;
+#
+# // Contents of _GLF_uniform_float_values: 1.0
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     float _GLF_uniform_float_values[1];
+# };
+# // Contents of _GLF_uniform_int_values: [5, 0, 20, 1, 2, 3, 4, 19, 16, 17, 18]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     int _GLF_uniform_int_values[11];
+# };
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     int arr[5] = int[5](_int_1, _int_2, _int_3, _int_4, _int_5);
+#     _GLF_color = vec4(_int_0);
+#
+#     // Iterated five times.
+#     for(int i = _int_0; i < _int_5; i ++)
+#     {
+#         int a = - 100;
+#         int b = 1;
+#
+#         // This loop has no effect.
+#         do
+#         {
+#             if(b > 7)
+#             {
+#                 break;
+#             }
+#
+#             switch(a)
+#             {
+#                 case 0:
+#                     // Never executed.
+#                     _GLF_color = vec4(_int_1);
+#                     break;
+#                 case 5:
+#                     // Never executed.
+#                     if(gl_FragCoord.x < _float_1_0)
+#                     {
+#                         _GLF_color = vec4(1);
+#                     }
+#                     break;
+#             }
+#
+#             a++;
+#             b++;
+#         }
+#         while(a < 10);
+#
+#         // Increments each array element by five.
```

Comment:
Is it by five, or by three? (I think the latter.)

Reply:
You're right :)